src/wp-includes/user.php

```diff
@@ -2460,8 +2460,8 @@ function wp_insert_user( $userdata ) {
 		 */
 		do_action( 'profile_update', $user_id, $old_user_data, $userdata );
 
-		if ( isset( $userdata['spam'] ) && $userdata['spam'] != $old_user_data->spam ) {
-			if ( 1 == $userdata['spam'] ) {
+		if ( $spam !== $old_user_data->spam ) {
+			if ( 1 === $spam ) {
```

Comment:
`$spam` is falsy `0` or a boolean value, so this should be `if ( true === $spam ) {`